driver/modern_bpf/helpers/store/auxmap_store_params.h

```diff
@@ -418,3 +439,314 @@ static __always_inline void auxmap__store_path_from_fd(struct auxiliary_map *aux
+		__builtin_memcpy(&ipv6, sockaddr_in6->sin6_addr.in6_u.u6_addr32, 16);
+		u16 port = sockaddr_in6->sin6_port;
+
+		/* Pack the sockaddr info:
+		 * - socket family.
+		 * - dest_ipv6.
+		 * - dest_port.
+		 */
+		push__u8(auxmap->data, &auxmap->payload_pos, socket_family_to_scap(socket_family));
+		push__ipv6(auxmap->data, &auxmap->payload_pos, ipv6);
+		push__u16(auxmap->data, &auxmap->payload_pos, ntohs(port));
+		final_param_len = FAMILY_SIZE + IPV6_SIZE + PORT_SIZE;
+		break;
+	}
+
+	case AF_UNIX:
+	{
+		/* Map the user-provided address to a sockaddr_un. */
+		struct sockaddr_un *sockaddr_un = (struct sockaddr_un *)sockaddr;
+
+		/* Starting at `sockaddr_un` we have the socket family and after it
+		 * the `sun_path`.
+		 *
+		 * Please note exceptions in the `sun_path`:
+		 * Taken from: https://man7.org/linux/man-pages/man7/unix.7.html
+		 *
+		 * An `abstract socket address` is distinguished (from a
+		 * pathname socket) by the fact that sun_path[0] is a null byte
+		 * ('\0').
+		 */
+
+		/* Check the exact point in which we have to start reading our path. */
+		unsigned long start_reading_point;
+		/* We skip the two bytes of socket family. */
+		char first_path_byte = *(char *)sockaddr_un->sun_path;
+		if(first_path_byte == '\0')
+		{
+			/* This is an abstract socket address, we need to skip the initial `\0`. */
+			start_reading_point = (unsigned long)sockaddr_un->sun_path + 1;
+		}
+		else
+		{
+			start_reading_point = (unsigned long)sockaddr_un->sun_path;
+		}
+
+		/* Pack the sockaddr info:
+		 * - socket family.
+		 * - socket_unix_path (sun_path).
+		 */
+		push__u8(auxmap->data, &auxmap->payload_pos, socket_family_to_scap(socket_family));
+		u16 written_bytes = push__charbuf(auxmap->data, &auxmap->payload_pos, start_reading_point, MAX_UNIX_SOCKET_PATH, KERNEL);
+		final_param_len = FAMILY_SIZE + written_bytes;
+		break;
+	}
+
+	default:
+		final_param_len = 0;
+		break;
+	}
+	push__param_len(auxmap->data, &auxmap->lengths_pos, final_param_len);
+}
+
+/**
+ * @brief Store socktuple info taken from kernel socket.
+ * We prefer extracting data directly from the kernel to
+ * obtain more precise information.
+ *
+ * Please note:
+ * In outbound connections `local` is the src while `remote` is the dest.
+ * In inbound connections vice versa.
+ *
+ * @param auxmap pointer to the auxmap in which we are storing the param.
+ * @param socket_fd socket from which we extract information about the tuple.
+ * @param direction specifies the connection direction.
+ */
+static __always_inline void auxmap__store_socktuple_param(struct auxiliary_map *auxmap, u32 socket_fd, int direction)
+{
+	u16 final_param_len = 0;
+
+	/* Get the socket family directly from the socket */
+	u16 socket_family = 0;
+	struct file *file = NULL;
+	file = extract__file_struct_from_fd(socket_fd);
+	struct socket *socket = BPF_CORE_READ(file, private_data);
+	struct sock *sk = BPF_CORE_READ(socket, sk);
+	BPF_CORE_READ_INTO(&socket_family, socket, ops, family);
```

Comment:
HI @Andreagit97 , I guess I got it:

With changing this to:
````c
BPF_CORE_READ_INTO(&socket_family, sk, __sk_common.skc_family);
````
tests passes on s390x:
````
[  PASSED  ] 83 tests.

# cat /sys/kernel/debug/tracing/trace
# tracer: nop
#
# entries-in-buffer/entries-written: 9/9   #P:2
#
#                                _-----=> irqs-off/BH-disabled
#                               / _----=> need-resched
#                              | / _---=> hardirq/softirq
#                              || / _--=> preempt-depth
#                              ||| / _-=> migrate-disable
#                              |||| /     delay
#           TASK-PID     CPU#  |||||  TIMESTAMP  FUNCTION
#              | |         |   |||||     |         |
        bpf_test-3015    [000] d....  1201.646247: bpf_trace_printk: socket_family: 0

        bpf_test-3015    [000] d....  1201.646630: bpf_trace_printk: socket_family: 0

        bpf_test-3015    [000] d....  1201.646992: bpf_trace_printk: socket_family: 0

        bpf_test-3171    [001] d....  1549.256385: bpf_trace_printk: socket_family: 2

        bpf_test-3171    [001] d....  1549.256447: bpf_trace_printk: socket_family: 10

        bpf_test-3171    [001] d....  1549.256516: bpf_trace_printk: socket_family: 1

        bpf_test-3174    [001] d....  1557.140907: bpf_trace_printk: socket_family: 2

        bpf_test-3174    [001] d....  1557.140982: bpf_trace_printk: socket_family: 10

        bpf_test-3174    [001] d....  1557.141097: bpf_trace_printk: socket_family: 1


````

Reply:
:heart_eyes: thank you very much you saved me a lot of time since I cannot test directly on `s390x` i will check if these changes are ok for `x86` and `aarch64`, hoping we can avoid conditional compilation :)

Reply:
> heart_eyes thank you very much you saved me a lot of time since I cannot test directly on `s390x`
If you like, you can sign up on the [LinuxONE community](https://linuxone.cloud.marist.edu/#/login) and get an `s390x` instance within a few minutes (and for some days). You likely need to build with bundled deps depending on the distribution level.

> i will check if these changes are ok for `x86` and `aarch64`, hoping we can avoid conditional compilation :)
Thanks!

Reply:
> If you like, you can register on the [LinuxONE community](https://linuxone.cloud.marist.edu/#/login) and get an s390x instance within a few minutes (and for some days). You likely need to build with bundled deps depending on the distribution level.

Wow i didn't know it, thank you it will be very useful!!

> Thanks! Actually, you pointed me into the [skc_*](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/net/sock.h#n184) direction later when extracting the destination address and ports.

Yeah probably we get the same information in different places but only on some architectures :thinking:

Reply:
> Yeah probably we get the same information in different places but only on some architectures thinking

Actually, kernel intializes the `sk->sk_family` [here](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/net/core/sock.c#n2031) and `sk_family` is a [macro](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/net/sock.h#n375) to map to `__sk_common.skc_family`. So hope that this works for all architecture. (However, it does not answer why extracting the family from the protocol ops structure returns zero on s390x.)

Reply:
@hbrueckner I've tested on `x86` and `aarch64` and it works smooth, thank you for the fix! :rocket: